frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -0,0 +1,45 @@
+import { useMemo, useRef } from 'react'
+
+import { PUBLICFORM_REGEX } from '~constants/routes'
+import { HttpError } from '~services/ApiService'
+
+import { PublicFormContext } from './PublicFormContext'
+import { usePublicFormView } from './queries'
+
+interface PublicFormProviderProps {
+  formId: string
+  children: React.ReactNode
+}
+
+export const PublicFormProvider = ({
+  formId,
+  children,
+}: PublicFormProviderProps): JSX.Element => {
+  const miniHeaderRef = useRef<HTMLDivElement>(null)
+
+  const { data, error, ...rest } = usePublicFormView(formId)
+
+  const isFormNotFound = useMemo(() => {
+    return (
+      !PUBLICFORM_REGEX.test(formId) ||
+      (error instanceof HttpError && error.code === 404)
+    )
+  }, [error, formId])
```

Comment:
Sorry, a little magic hidden in `usePublicFormView()` in that the same check is ALSO done in that hook, in that the query doesn't fire if the regex does not match:

```ts
const usePublicFormView = (formId) => {
  return useQuery(
    publicFormKeys.id(formId),
    () => getPublicFormView(formId),
    {
      enabled: PUBLICFORM_REGEX.test(formId), // <------
      refetchOnWindowFocus: false,
    },
  )
}
```

However, the hook is not responsible for rendering a 404 page (and only responsible for the API call) and thus the same check is performed in `PublicFormProvider`.

Reply:
> * Since `error` is retrieved by querying on `formId`, `error` can't change by itself and so is not a true dependency of this memoization, only `formId` is

It is still a dependency of the memoization in that we want the memoized variable to change whenever error changes

Reply:
Right. Just for my understanding: what is the execution flow here that causes the API call to be reissued on the same formID?

Reply:
on the small off chance that our server does not return a `200` status (or if user has no internet), `react-query` will retry the query. That's the only place where it may happen.